src/switchroot/ostree-prepare-root.c

```diff
@@ -393,11 +448,8 @@ main (int argc, char *argv[])
       // however, see
       // https://github.com/systemd/systemd/blob/604b2001081adcbd64ee1fbe7de7a6d77c5209fe/src/basic/mountpoint-util.h#L36
       // which bumps up these defaults for the rootfs a bit.
-      g_autofree char *root_upperdir
-          = root_transient ? g_build_filename (OTCORE_RUN_OSTREE_PRIVATE, "root/upper", NULL)
-                           : NULL;
-      g_autofree char *root_workdir
-          = root_transient ? g_build_filename (OTCORE_RUN_OSTREE_PRIVATE, "root/work", NULL) : NULL;
+      const char *root_upperdir = OTCORE_RUN_OSTREE_PRIVATE "/root/upper";
```

Comment:
Wait isn't this going to turn on `root.transient` by default?

Reply:
These variables are not used if `root_transient` is false. The code is misleading. I'll put these variable definitions into the `if` block.